src/main/java/io/redisearch/client/Client.java

```diff
@@ -519,4 +432,180 @@ public boolean dropIndex(boolean missingOk) {
         }
         return r.equals("OK");
     }
+
+    @Override
+    public Long addSuggestion(Suggestion suggestion, boolean increment) {
+        ArrayList<String> args = new ArrayList(
+                Arrays.asList(this.indexName, suggestion.getString(), Double.toString(suggestion.getScore())));
+
+        if (increment) {
+            args.add(INCREMENT_FLAG);
+        }
+        if (suggestion.getPayload() != null) {
+            args.add(PAYLOAD_FLAG);
+            args.add(suggestion.getPayload());
+        }
+
+        try (Jedis conn = _conn()) {
+            return sendCommand(conn, AutoCompleter.Command.SUGADD, args.toArray(new String[args.size()])).getIntegerReply();
+        }
+    }
+
+    @Override
+    public List<Suggestion> getSuggestion(String prefix, SuggestionOptions suggestionOptions) {
```

Comment:
I'd still prefer this be branched out to other smaller helper methods. I don't like the big API, but internally we can branch however we'd like.